### PR TITLE
feat: add translucent status bar handling

### PR DESCRIPTION
**Pull request by WoLewicki**
Added a listener on the insets of the window which mimics the behavior of RN's StatusBar: (https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java#L145). The addition of padding is controlled by the exported `translucentStatusBar` prop. The prop is needed because of the incorrect behavior which occurred sometimes if the padding was only controlled by the listener, where the toolbar's top padding was added to the window insets top padding, causing the doubled top padding of the toolbar.
The situation handled here is the case where a user has his status bar set to translucent via RN StatusBar component with `translucent={true}` prop. In that case, the header's top padding is not applied properly which causes the content of the header to go under the Status Bar. See #485 for more info.

### CODE REVIEW

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java [hunk 1]
```diff
@@ -302,6 +312,23 @@ private TextView getTitleTextView() {
...
+      @Override
+      public WindowInsets onApplyWindowInsets(View v, WindowInsets insets) {
+        WindowInsets defaultInsets = v.onApplyWindowInsets(insets);
+        // this inset is set to 0 by translucent status bar
+        if (defaultInsets.getStableInsetTop() == 0) {
+          mToolbar.setPadding(0, insets.getSystemWindowInsetTop(), 0, 0);
+        } else {
+          mToolbar.setPadding(0, defaultInsets.getSystemWindowInsetTop(), 0, 0);
+        }
+        return insets.consumeSystemWindowInsets();
```
Comment (mczernek) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:327]: You really consumed only top inset in here, haven't you? There should be a way of marking it.

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java [hunk 2]
```diff
@@ -173,6 +178,11 @@ public void onUpdate() {
     mToolbar.setNavigationOnClickListener(mBackClickListener);
 
 
+    // we apply padding if status bar is translucent
```
Comment (mczernek) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:181]: You could use same constant as in `RequiresApi` annotation in `applyWindowInsets` method

**Comment by janicduplessis** [comment]
Does this work if translucent was set using system ui visibility flags instead of using the RN statusbar module?

```
getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
```

**Comment by WoLewicki** [comment]
@janicduplessis I changed the behavior to mimic what the RN StatusBar does: https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java#L119. It requires the user to apply a prop if he wants it to work correctly, but it eliminates the problems with additional padding added which sometimes occurred with the previous implementation. It should also work ok with system ui visibility flags, because you can now control the padding by yourself with the prop.

**Comment by janicduplessis** [comment]
:+1: using a prop seems like a good approach to me.

**Comment by Hirbod** [comment]
Maybe it should be called translucentStatusBarAndroid or something. That wording could be confusing

**Comment by WoLewicki** [comment]
@Hirbod it has mentions of the platform in the types definition files. I think it should be enough since it is the convention in this library.

File: native-stack/types.tsx [hunk 3]
```diff
@@ -210,6 +210,13 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   gestureEnabled?: boolean;
+  /**
+   * Set this prop to true whenever you have translucent status bar in order for it to work correctly. Defaults to `false`.
+   * Only supported on Android.
+   *
+   * @platform android
+   */
+  translucentStatusBar?: boolean;
```
Comment (brentvatne) [inline native-stack/types.tsx:219]: `insetForTranslucentStatusBar` maybe? otherwise it feels like we are setting the status bar to translucent

**Comment by WoLewicki** [comment]
Closing due to #545.

**Pull request closed without merging**